### PR TITLE
Add extensions for getting a MemberName from a ClassName, KClass and Class

### PR DESCRIPTION
**Pull request by ansman**

### CODE REVIEW

**Comment by ansman** [comment]
Added

File: src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt [hunk 1]
```diff
@@ -216,4 +217,40 @@ class MemberNameTest {
...
+        .isEqualTo(MemberName("kotlin.text", regexCompanion, "fromLiteral"))
+  }
+
+  @Test fun memberExtension_kclass() {
+    val regex = ClassName("kotlin.text", "Regex")
+    assertThat(Regex::class.member("fromLiteral"))
+        .isEqualTo(MemberName("kotlin.text", regex, "fromLiteral"))
+  }
+
+  @Test fun memberExtension_kclass_nested() {
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt]: Same for this one

File: src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt [hunk 2]
```diff
@@ -216,4 +217,40 @@ class MemberNameTest {
...
+  }
+
+  @Test fun memberExtension_class() {
+    val regex = ClassName("kotlin.text", "Regex")
+    assertThat(Regex::class.java.member("fromLiteral"))
+        .isEqualTo(MemberName("kotlin.text", regex, "fromLiteral"))
+  }
+
+  @Test fun memberExtension_class_nested() {
+    val regexCompanion = ClassName("kotlin.text", "Regex", "Companion")
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt]: And this one

File: src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt [hunk 3]
```diff
@@ -216,4 +217,40 @@ class MemberNameTest {
...
       |""".trimMargin())
   }
+
+  @Test fun memberExtension_className() {
+    val regex = ClassName("kotlin.text", "Regex")
+    assertThat(regex.member("fromLiteral"))
+        .isEqualTo(MemberName("kotlin.text", regex, "fromLiteral"))
+  }
+
+  @Test fun memberExtension_className_nested() {
```
[the remark this answers is not in the document]
Comment (ansman) [reply src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt]: Well, it tests that a nested class is handled correctly, but I can remove it if you don't think it's useful

**Comment by Egorand** [comment]
Thanks!

**Pull request merged**